datasetinsights/datasets/groceries_real.py

```diff
@@ -217,6 +214,39 @@ def _convert_to_bbox2d(single_bbox):
         )
         return canonical_bbox
 
+    @staticmethod
+    def _preprocess_dataset(data_path):
+        # check if compressed dataset file is present
+        if os.path.isfile(os.path.join(data_path, "dataset")):
+            unzip_folder = os.path.join(data_path, "groceries_real")
+            decompress(
+                filepath=os.path.join(data_path, "dataset"),
+                destination=unzip_folder,
+            )
+
+            # check if necessary files are present after decompression
+            if GroceriesReal.is_dataset_files_present(unzip_folder):
+                return unzip_folder
+            else:
+                raise DatasetNotFoundError(
+                    f"No dataset file(s) present at path" f":{unzip_folder}"
+                )
+
+        # check if dataset files are present
+        if GroceriesReal.is_dataset_files_present(data_path):
```

Comment:
Should this method call be before the check to see if the compressed dataset file is present?